python/plugins/processing/gui/Postprocessing.py

```diff
@@ -289,11 +289,22 @@ def handleAlgorithmResults(
                 current_node_index = current_node_group.children().index(
                     current_selected_node
                 )
-                current_node_group.insertChildNode(current_node_index, layer_node)
+                insertion_point = QgsLayerTreeRegistryBridge.InsertionPoint(
+                    current_node_group, current_node_index
+                )
             elif isinstance(current_selected_node, QgsLayerTreeGroup):
-                current_selected_node.insertChildNode(0, layer_node)
+                insertion_point = QgsLayerTreeRegistryBridge.InsertionPoint(
+                    current_selected_node, 0
+                )
             elif context.project():
-                context.project().layerTreeRoot().insertChildNode(0, layer_node)
+                insertion_point = QgsLayerTreeRegistryBridge.InsertionPoint()
+
+        if insertion_point:
+            details.project.layerTreeRegistryBridge().setLayerInsertionPoint(
+                insertion_point
+            )
```

Comment:
This change would need resetting afterwards, otherwise the project will behave oddly when new layers are added outside of processing.

Also, looking at how the insertion point logic works in the layer tree bridge there's a big risk of crashes! I'll try to fix those now...

Reply:
> This change would need resetting afterwards, otherwise the project will behave oddly when new layers are added outside of processing.

I have added that. However, I have realized that there is no `LayerInsertionPoint` getter. So, I have added one.

Reply:
> Also, looking at how the insertion point logic works in the layer tree bridge there's a big risk of crashes! I'll try to fix those now...

Do you mean https://github.com/qgis/QGIS/pull/60446 ?